Objects/genobject.c

```diff
@@ -1161,17 +1162,14 @@ PyCoro_New(PyFrameObject *f, PyObject *name, PyObject *qualname)
     int origin_depth = tstate->coroutine_origin_tracking_depth;
 
     if (origin_depth == 0) {
-        ((PyCoroObject *)coro)->cr_origin = NULL;
-    } else {
-        PyObject *cr_origin = compute_cr_origin(origin_depth);
-        if (!cr_origin) {
-            Py_DECREF(coro);
-            return NULL;
-        }
-        ((PyCoroObject *)coro)->cr_origin = cr_origin;
```

Comment:
The PR can be reduced to very trivial change: move this line before `if (!cr_origin) {`, do nothing else

Reply:
Sure, but I had wanted to use the opportunity to simplify the code.